crates/biome_js_analyze/src/lint/nursery/use_explicit_function_return_type.rs

```diff
@@ -0,0 +1,158 @@
+    /// // Should indicate that no value is returned (void)
+    /// function test(a: number) {
+    ///   a += 1;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    /// ```ts
+    /// // No return value should be expected (void)
+    /// function test(): void {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// // A return value of type number
+    /// var fn = function (): number {
+    ///   return 1;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// // A return value of type string
+    /// var arrowFn = (): string => 'test';
+    /// ```
+    ///
+    /// ```ts
+    /// class Test {
+    ///   // No return value should be expected (void)
+    ///   method(): void {
+    ///     return;
+    ///   }
+    /// }
+    /// ```
+    ///
+    pub UseExplicitFunctionReturnType {
+        version: "next",
+        name: "useExplicitFunctionReturnType",
+        language: "ts",
+        recommended: false,
+        sources: &[RuleSource::EslintTypeScript("explicit-function-return-type")],
+    }
+}
+
+declare_node_union! {
+    pub AnyJsFunctionAndMethod = AnyJsFunction | JsMethodClassMember | JsGetterClassMember
+}
+
+impl Rule for UseExplicitFunctionReturnType {
+    type Query = Ast<AnyJsFunctionAndMethod>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        match node {
+            AnyJsFunctionAndMethod::AnyJsFunction(func) => {
+                if func.return_type_annotation().is_some() {
+                    return None;
+                }
+
+                let func_range = func.syntax().text_range();
+                if let Ok(Some(AnyJsBinding::JsIdentifierBinding(id))) = func.id() {
+                    return Some(TextRange::new(
+                        func_range.start(),
+                        id.syntax().text_range().end(),
+                    ));
+                }
+
+                Some(func_range)
+            }
+            AnyJsFunctionAndMethod::JsMethodClassMember(method) => {
+                if method.return_type_annotation().is_some() {
+                    return None;
+                }
+
+                Some(method.node_text_range())
+            }
+            AnyJsFunctionAndMethod::JsGetterClassMember(getter) => {
+                if getter.return_type().is_some() {
+                    return None;
+                }
+
+                Some(getter.node_text_range())
+            }
+        }
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                state,
+                markup! {
+                    "Missing return type on function."
+                },
+            )
+            .note(markup! {
+                "Require explicit return types on functions and class methods."
```

Comment:
This note looks a bit redundant to me.
We should follow our [rule pillars](https://biomejs.dev/linter/#rule-pillars) and explain why a user should add a return type.

Something like:

`Declaring the return type self-documents the code and can speed up TypeScript type checking.`

We could also add an extra note telling the user to add a return type `Add a return type annotation`.

Reply:
That makes sense. I’ve updated the note accordingly!!